onyo/commands/new.py

```diff
@@ -21,86 +21,108 @@
 reserved_characters = ['_', '.']
 
 
-def build_commit_cmd(file, onyo_root):
-    return ["git -C \"" + onyo_root + "\" commit -m", "new asset.\n\n" + file]
-
-
 def read_new_word(word_description):
+    """
+    Read a new word via user input for the creation of a asset name.
+    Checks for the serial field if it contains reserved characters, and checks
+    for all fields that they are not empty. If a check is not successful, it
+    tries to read the same word again.
+
+    Returns the word on success.
+    """
     # read word for field from keyboard
-    word = str(input(word_description))
+    word = input(word_description)
+
     # if word contains reserved character, inform and read new word
     for char in reserved_characters:
         if word_description != "<serial>*:" and char in word:
-            logger.info(char + " is in list of reserved characters: " + ", ".join(reserved_characters))
+            logger.info(f"'{char}' is in list of reserved characters: {reserved_characters}")
             return read_new_word(word_description)
+
     # if enter pressed without input, read new word
     if len(word) == 0:
         return read_new_word(word_description)
     return word
 
 
-def run_onyo_new(directory, template, non_interactive, onyo_root):
-    # create an asset file, first without content
+def run_onyo_new(directory, template, non_interactive, onyo_root, repo):
+    """
+    Read a new asset name, check it for uniqueness and validity, create the
+    asset (from a template if one was given), edit the new asset (if not
+    suppressed with flag) and stage it with git.
+
+    Returns the newly created asset's filename on success.
+    """
+    # create an asset file
     filename = create_filename(onyo_root, template)
-    if os.path.exists(os.path.join(directory, filename)):
-        logger.error(os.path.join(directory, filename) + " asset already exists.")
+    filename = Path(directory, filename).resolve()
+    template = Path(template).resolve()
+
+    if filename.exists():
+        logger.error(f"{filename}' asset already exists.")
         sys.exit(1)
-    if template:
-        run_cmd("cp \"" + template + "\" \"" + os.path.join(os.path.join(onyo_root, directory), filename) + "\"")
+    if template.is_file():
+        shutil.copyfile(template, filename)
     else:
-        run_cmd(create_asset_file_cmd(directory, filename))
+        filename.touch()
+
     # open new file?
     if not non_interactive:
```

Comment:
No need to change it for this PR, but double negatives are hard to read. Take a look at what I did to `onyo history` and how it uses `interactive` as the destvar.